src/dist/block_service/fds/fds_service.cpp

```diff
@@ -95,10 +99,46 @@ DEFINE_THREAD_POOL_CODE(THREAD_POOL_FDS_SERVICE)
 DEFINE_TASK_CODE(LPC_FDS_CALL, TASK_PRIORITY_COMMON, THREAD_POOL_FDS_SERVICE)
 
 const std::string fds_service::FILE_LENGTH_CUSTOM_KEY = "x-xiaomi-meta-content-length";
-const std::string fds_service::FILE_LENGTH_KEY = "content-length";
 const std::string fds_service::FILE_MD5_KEY = "content-md5";
 
-fds_service::fds_service() {}
+fds_service::fds_service()
+{
+    const int BYTE_TO_BIT = 8;
+
+    /// In normal scenario, the sst file size of level 0 is write_buffer_size * [0.75, 1.25]
+    /// And in BULK_LOAD scenario, it is 4 * write_buffer_size * [0.75, 1.25].
+    /// In rdsn, we can't get the scenario, so if we take BULK_LOAD scenario into consideration,
+    /// we must set max_sst_file_size to 4 * write_buffer_size * [0.75, 1.25], which is too big.
+    /// So in this implementation, we don't take BULK_LOAD scenario into consideration.
+    uint64_t target_file_size =
+        dsn_config_get_value_uint64("pegasus.server",
+                                    "rocksdb_target_file_size_base",
+                                    64 * 1024 * 1024,
+                                    "rocksdb options.target_file_size_base");
+    uint64_t write_buffer_size = dsn_config_get_value_uint64("pegasus.server",
+                                                             "rocksdb_write_buffer_size",
+                                                             64 * 1024 * 1024,
+                                                             "rocksdb options.write_buffer_size");
+    uint64_t max_sst_file_size = std::max(target_file_size, (uint64_t)1.25 * write_buffer_size);
+
+    uint32_t write_rate_limit = (uint32_t)dsn_config_get_value_uint64(
+        "replication", "fds_write_limit_rate", 20, "rate limit of fds(Mb/s)");
```

Comment:
consider add `remote_command` for the config?

Reply:
Runtime update of configurations should be a basic function of our config utility. We don't have to support dynamic updates one by one and case by case.

Reply:
Maybe we can implement runtime-update of configurations as a basic function of our config utility. We don't have to support dynamic updates one by one and case by case.

Reply:
> Maybe we can implement runtime-update of configurations as a basic function of our config utility. We don't have to support dynamic updates one by one and case by case.

Yes, I think it is a good idea.